Utils/API/server/lib/dkb/api/storages/es/methods.py

```diff
@@ -0,0 +1,415 @@
+"""
+Interaction with DKB ES storage.
+"""
+
+import logging
+import json
+from datetime import datetime
+import time
+
+from api.exceptions import MethodException
+from api.storages.exceptions import NoDataFound
+from api.common import DATE_FORMAT
+
+from . import STORAGE_NAME
+import common
+from common import (TASK_KWARGS,
+                    WARNINGS)
+from common import (init,
+                    client,
+                    task_info,
+                    output_formats,
+                    tokens,
+                    get_query,
+                    get_selection_query,
+                    get_step_aggregation_query,
+                    get_agg_units_query)
+import transform
+
+
+def task_steps_hist(**kwargs):
+    """ Implementation of :py:func:`storages.task_steps_hist` for ES.
+
+    :return: hash with histogram data
+    :rtype: dict
+    """
+    init()
+    q = dict(TASK_KWARGS)
+    start = kwargs.get('start')
+    end = kwargs.get('end')
+    if end:
+        current_ts = end
+    else:
+        current_ts = datetime.utcnow()
+    kwargs['current_ts_ms'] = int(time.mktime(current_ts.timetuple())) * 1000
+    q['body'] = get_query('task-steps-hist', **kwargs)
+    if start:
+        r = {'range': {'end_time': {'gte': start.strftime(DATE_FORMAT)}}}
+        q['body']['query']['bool']['must'].append(r)
+    if end:
+        r = {'range': {'start_time': {'lte': end.strftime(DATE_FORMAT)}}}
+        q['body']['query']['bool']['must'].append(r)
+    data = client().search(**q)
+    result = transform.task_steps_hist(data, start, end)
+    return result
+
+
+def task_chain(**kwargs):
+    """ Reconstruct task chain from given task ID.
+
+    If task not found in the ES, raises ``NoDataFound``.
+
+    If no ``chain_id`` was found for given task, its taskiID is used instead.
```

Comment:
A suggested change was made here, as well as some other improvements, but now there is a "taskiID".

Reply:
Right, sorry for being inobservant. Fixed now: #373.